crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -205,6 +205,7 @@ where
     'b: 'a,
 {
     fn visit_stmt(&mut self, stmt: &'b Stmt) {
+        // Phase 0: Pre-processing
```

Comment:
Can we add some documentation to `Checker` that explains the different phases? Especially `Analysis` is unclear to me (`Syntax`?)